app/controllers/concerns/part_of_claim_journey.rb

```diff
@@ -25,23 +25,41 @@ def send_unstarted_claimants_to_the_start
     redirect_to current_policy.start_page_url unless current_claim.persisted?
   end
 
+  # Keep this for StudentLoans and MathsAndPhysics journeys
+  # Keep ECP journey working for now with current_claim as we build out combined journey
   def current_claim
     @current_claim ||= claim_from_session || build_new_claim
   end
 
+  # Placeholder needs session work
+  def current_ecp_claim
+    @current_ecp_claim
+  end
+
+  # Placeholder needs session work
+  def current_lup_claim
+    @current_lup_claim
+  end
+
+  # This will need to manage multiple claims
   def claim_from_session
     Claim.find(session[:claim_id]) if session.key?(:claim_id)
   end
 
   def build_new_claim
-    Claim.new(
-      eligibility: current_policy::Eligibility.new,
-      academic_year: policy_configuration.current_academic_year
-    )
+    Journeys.policies_for_routing_name(current_policy_routing_name).map do |policy|
+      claim = Claim.new(
+        eligibility: policy::Eligibility.new,
+        academic_year: policy_configuration.current_academic_year
+      )
+      @current_ecp_claim = claim if policy == EarlyCareerPayments
+      @current_lup_claim = claim if policy == LevellingUpPayments
+      claim
+    end.first
```

Comment:
Again making the `claims` more generic and an Array so we're not hard-coding references to the two policies.

```suggestion
    @claims = Journeys.policies_for_routing_name(current_policy_routing_name).map do |policy|
      claim = Claim.new(
        eligibility: policy::Eligibility.new,
        academic_year: policy_configuration.current_academic_year
      )
      claim
    end.first
```